src/docs/background/16-how-saving-payment-card-works-in-ftw/index.md

```diff
@@ -0,0 +1,135 @@
+published: true
+---
+
+## Introduction
+
+When a customer first comes to the marketplace and books a listing,
+there are several form fields that need to be updated: card number, its
+expiration month, card verification code (CVC), card holder's name and
+possibly other billing details. To improve user experience for returning
+customers, it's good to have an option to save payment card details for
+future bookings. If there is an existing payment card available, the
+user can just click the "Send request" button to complete checkout page.
+
+From that point forward, the typical case is that customer pays upfront
+(aka makes an
+[on-session](https://stripe.com/docs/payments/cards/reusing-cards#charging-on-session)
+payment). In practice, a preauthorization is created: the money is
+reserved, but not yet moved to Stripe. When a provider accepts the
+request, the preauthorization is captured (i.e. the payment is charged
+from the card and money moved to Stripe and held on the connected
+account of the provider until payout). The actual payout happens when
+the transaction completes.
+
+Payment card needs also be saved for off-session payments if those are
+in use. They are automatic one-time payments that happen when the user
+is not interacting with your application. You can read more about
+off-session payments from a separate
+[article](/background/off-session-payments-in-transaction-process).
+
+## Saving card details
+
+There are 2 ways to save payment card details:
+
+- saving cards details without making an initial payment
+- saving cards details after a payment
+
+The former of those can be done in payment methods page (profile menu ->
+account settings -> payment methods), but you can also save payment card
+details on the checkout page when making a one-time payment.
+
+> Note: currently, it is only possible to save one card, which becomes
+> the default payment method. I.e. if there is already one payment card
+> saved, your only option is to replace the card with a new one.
+
+### Saving cards without making an initial payment
+
+If the user has no existing default payment card, `PaymentMethodsPage`
+component will show just a form to save payment card details:
+
+![Payment methods page](payment-methods-page.png 'Payment methods page without default payment method.')
+
+If there is a default payment methods saved, user will see a dropdown
+instead, where the current default method is selected. There are two
+options the user can do: replace the current payment card (it's an
+option inside the dropdown) or delete the current payment card.
+
+![Payment methods page saved card](payment-methods-page-saved-card.png 'Payment methods page with default payment method.')
+
+Under the hood, saving a new payment method needs to complete following
+steps:
+
+- Create Stripe
+  [Setup Intent](https://stripe.com/docs/payments/cards/saving-cards#create-setup-intent)
+  through Flex API to obtain the client secret.
```

Comment:
Maybe we should add link to our API reference about creating the setup intent just to make sure it's clear enough that it's handled through our API.

Also, I don't know would it be useful to add links to API reference or e.g. in related duck files so that related code examples would be easier to find.

Reply:
I'll assume that the link will be:
https://www.sharetribe.com/api-reference/index.html#stripe-setup-intents
:)